src/org/jitsi/impl/neomedia/ProcessorUtility.java

```diff
@@ -136,18 +144,27 @@ else if (state == Processor.Realized)
 
         boolean interrupted = false;
 
-        // Wait until we get an event that confirms the
-        // success of the method, or a failure event.
-        // See StateListener inner class
-        while ((processor.getState() < state) && !failed)
+        Object stateLock = getStateLock();
+        synchronized (stateLock)
         {
-            Object stateLock = getStateLock();
-
-            synchronized (stateLock)
+            // Wait until we get an event that confirms the
+            // success of the method, or a failure event.
+            // See StateListener inner class
+            while ((processor.getState() < state) && !failed)
             {
                 try
                 {
-                    stateLock.wait();
+                    // don't wait forever, there is some other
+                    // problem where we wait on an already closed
+                    // processor and we never leave this wait
+                    long startTime = System.nanoTime();
+                    stateLock.wait(WAIT_TIMEOUT);
+                    if (System.nanoTime() - startTime
+                            > TimeUnit.MILLISECONDS.toNanos(WAIT_TIMEOUT))
```

Comment:
Since we're on Java 8 now, why don't you use `Instant` and `Duration` instead of juggling manual date conversions?

Reply:
I just updated it, does it look ok?